rfcs/convergence/first-rule-of-aria.md

```diff
@@ -0,0 +1,164 @@
+- A `<select>` element cannot be recreated with ARIA in a way that works seamlessly across platforms: mobile browsers often have completely custom rendering of the options list, and the platform API mappings are actually different between Windows, macOS, and iOS/Android.
+- Sometimes live regions are used to make up for differences between native behavior and custom ARIA widgets. This technique usually falls spectacularly flat on braille displays, where live regions have notably flaky support, particularly when they conflict with interaction.
+
+The difficulty replicating the robust nature of HTML with custom scripting and ARIA is why the [First Rule of ARIA](https://w3c.github.io/using-aria/#rule1) is "don't use ARIA".
+
+## Detailed Proposal
+
+The proposal is a pattern in favor of using semantic elements as a default (with opt-out mechanisms) for convergence components. This is particularly important for interactive elements like buttons, links, inputs, radios, etc.
+
+This proposal does not introduce any API, is just a pattern to be followed when creating converged components, and valid to add that in extreme cases, like table, that semantic first approach is simply not viable, than this should be ignored.
+
+Converged components should follow these steps:
+
+---
+
+If is equivalent to native element - `Button`, `Link`, `Select`
+
+- Should use native element by default
+- Should implement [as-props](./as-props.md) to opt-out native element
+- Should limit possible elements to opt-out (e.g: `Button` should opt-out from `button` for `div` and `span`)
+- In case of opting-out element should implement required ARIA event listeners and attributes, such as `role`
+- Provide authors an option to use a semantic HTML version (e.g. Dropdown should have an author option to use a `<select>` element, even if we also allow a custom version with a stylable extendable options list).
+
+##### Button Example
+
+```tsx
+<Button>This is a Simple Button</Button>
+<Button as="div">This is a div that looks and behaves as a Button</Button>
+<Button as="div" role={undefined} tabIndex={undefined}>
+  This is a div that looks like Button but doesn't behave as such
+  therefore, I can add a <Link to="/somewhere">Link inside of it</Link>
+</Button>
+```
+
+```html
+<!-- Semantic version -->
+<button class="button-class">This is a Simple Button</button>
+<!-- Aria version -->
+<div class="button-class" tabindex="0" role="button">This is a div that looks and behaves as a Button</div>
+<!-- Style only -->
+<div class="button-class">
+  This is a div that looks like Button but doesn't behave as such therefore, I can add a
+  <a href="/somewhere">Link inside of it</a>
+</div>
+```
+
+---
+
+If isn't equivalent to native elements but can benefit from using native elements in it's implementation - `Accordion`, `Carousel`, `Disclosure`, `MenuButton`, `Menu`, `MenuBar`, `Tabs`
```

Comment:
Can you make it clearer what these benefits are ?

Reply:
I would prefer for this to be `role` oriented, because if the `role` differs from the semantic HTML element it might not be very useful if you would need to reimplement the behaviour of those native elements.

Reply:
Some of the benefits of using a `<button>` for tabs, accordion triggers, menu buttons, etc. include high contrast styles by default and keyboard invocation by default. There may be other benefits to using a natively interactive element that I'm not aware of, e.g. in assistive tech we don't test for.

Reply:
But I do agree there are some limits (e.g. `<button>` should only be used for controls that perform an action on click), and in the case of components like Menu, the benefit is likely less because we're already managing keyboard interaction, and button HCM colors aren't as important for menuitems as for standalone buttons.

Reply:
I didn't think about HCM colors, that's a good call out, let's add the benefits (i.e. interaction + HCM colors) in this RFC